tests/core/test_addasset.py

```diff
@@ -0,0 +1,74 @@
+import os
+from tempfile import TemporaryDirectory
+from unittest import TestCase
+
+import pystac
+
+from stactools.core import add_asset_to_item
+from tests import test_data
+from tests.utils import create_temp_copy
+
+
+class AddAssetTest(TestCase):
```

Comment:
Thanks for adding some `core` unit tests! I think there are a couple more cases that would be useful to assert:

1. Asserting that an error is raised if `move=True` and the item doesn't have an href
2. Asserting that things work ok if the item doesn't have an href, `move=False`, and the asset href is absolute
3. Asserting an error if the item doesn't have an href, `move=False`, and the asset href is relative

There may be a couple others. When reading your tests, I was curious what happened in the above cases, and had to go read the code to find out.

Reply:
I have added tests for these cases and a few more.